rl4co/envs/scheduling/ffsp.py

```diff
@@ -333,11 +421,18 @@ def generate_data(self, batch_size) -> TensorDict:
         batch_size = [batch_size] if isinstance(batch_size, int) else batch_size
 
         # Init observation: running time of each job on each machine
-        run_time = (
-            torch.FloatTensor(*batch_size, self.num_job, self.num_machine, self.num_stage)
-            .uniform_(self.min_time, self.max_time)
-            .to(self.device)
-        )
+        run_time = torch.randint(
+            low=self.min_time,
+            high=self.max_time,
+            size=(*batch_size, self.num_job, self.num_machine, self.num_stage),
+        ).to(self.device)
+
+        if self.flatten_stages:
+            run_time = (
+                run_time.transpose(-2, -1)
+                .contiguous()
+                .view(*batch_size, self.num_job, self.num_machine_total)
```

Comment:
[Comment only] Great, these `view` parts are always tricky... I wish there was a standardized way to do this